components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementFacade.java

```diff
@@ -0,0 +1,447 @@
+            throws IdentityProviderManagementException {
+
+        String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
+        addEndpointConfig(identityProvider, tenantDomain);
+        try {
+            return dao.addIdPWithResourceId(identityProvider, tenantId);
+        } catch (IdentityProviderManagementException e) {
+            deleteEndpointConfig(identityProvider, tenantDomain);
+            throw e;
+        }
+    }
+
+    public void updateIdP(IdentityProvider newIdentityProvider, IdentityProvider currentIdentityProvider, int tenantId)
+            throws IdentityProviderManagementException {
+
+        updateEndpointConfig(newIdentityProvider, currentIdentityProvider,
+                IdentityTenantUtil.getTenantDomain(tenantId));
+        try {
+            dao.updateIdP(newIdentityProvider, currentIdentityProvider, tenantId);
+        } catch (IdentityProviderManagementException e) {
+            updateEndpointConfig(currentIdentityProvider, newIdentityProvider,
+                    IdentityTenantUtil.getTenantDomain(tenantId));
+            throw e;
+        }
+    }
+
+    public void updateIdPWithResourceId(String resourceId, IdentityProvider newIdentityProvider,
+                                        IdentityProvider currentIdentityProvider, int tenantId)
+            throws IdentityProviderManagementException {
+
+        updateEndpointConfig(newIdentityProvider, currentIdentityProvider,
+                IdentityTenantUtil.getTenantDomain(tenantId));
+        try {
+            dao.updateIdPWithResourceId(resourceId, newIdentityProvider, currentIdentityProvider, tenantId);
+        } catch (IdentityProviderManagementException e) {
+            updateEndpointConfig(currentIdentityProvider, newIdentityProvider,
+                    IdentityTenantUtil.getTenantDomain(tenantId));
+            throw e;
+        }
+    }
+
+    public boolean isIdpReferredBySP(String idPName, int tenantId) throws IdentityProviderManagementException {
+
+        return dao.isIdpReferredBySP(idPName, tenantId);
+    }
+
+    public void deleteIdP(String idPName, int tenantId, String tenantDomain)
+            throws IdentityProviderManagementException {
+
+        IdentityProvider identityProvider = getIdPByName(null, idPName, tenantId, tenantDomain);
+        deleteEndpointConfig(identityProvider, tenantDomain);
+        try {
+            dao.deleteIdP(idPName, tenantId, tenantDomain);
+        } catch (IdentityProviderManagementException e) {
+            addEndpointConfig(identityProvider, tenantDomain);
+            throw e;
+        }
+    }
+
+    public void deleteIdPs(int tenantId) throws IdentityProviderManagementException {
+
+        List<IdentityProvider> idpList = getIdPs(null, tenantId, IdentityTenantUtil.getTenantDomain(tenantId));
+        for (IdentityProvider idp : idpList) {
+            deleteEndpointConfig(idp, IdentityTenantUtil.getTenantDomain(tenantId));
+        }
```

Comment:
IMO, we should have a way to delete endpointis in a batch not in a loop.

Also adding should happen in a batch

Reply:
Executing this in a batch could be difficult since there are set of operations needed to do in ActionManagementService such as encrypting/decrypting secrets.

Reply:
Yes. But it includes deleting secrets.

Reply:
@Thisara-Welmilla as discussed offline let's track this improvement and better add a comment as a todo

Reply:
This will be tracked with https://github.com/wso2/product-is/issues/21783